### PR TITLE
`azurerm_spatial_anchors_account` - support for `account_domain` and `account_id`

### LINKED ISSUE #11799: Return account_domain account_id from spatial_anchors_account properties + Add Data Source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Mixed Reality/Spatial Anchors Account should return account_domain and account_id from properties object belonging to the created Spatial Anchors Account. It should also be possible to retrieve this information with a Data source.

I see that that the Go client package currently used is  `preview/mixedreality/mgmt/2019-02-28/mixedreality`. 
Might be necessary to use `mixedreality/mgmt/2021-01-01/mixedreality` instead to get the properties. 

### New or Affected Resource(s)

* azurerm_spatial_anchors_account (resource, affected)
* azurerm_spatial_anchors_account (data, new)

### Potential Terraform Configuration

```hcl
data "azurerm_spatial_anchors_account " "example" {
}
```

### References

* #0000

### PR DESCRIPTION
**Pull request by favoretti**
* deps: upgrade `mixedreality` to `2021-01-01`
* New data source: `azurerm_spatial_anchors_account`
* Add `account_id` and `account_domain` properties to
  `azurerm_spatial_anchors_account` resource

```
$ TF_ACC=1 go test -v ./azurerm/internal/services/mixedreality -timeout=1000m
2021/05/22 11:25:31 [DEBUG] not using binary driver name, it's no longer needed
2021/05/22 11:25:31 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccSpatialAnchorsAccountDataSource_basic
=== PAUSE TestAccSpatialAnchorsAccountDataSource_basic
=== RUN   TestAccSpatialAnchorsAccount_basic
=== PAUSE TestAccSpatialAnchorsAccount_basic
=== RUN   TestAccSpatialAnchorsAccount_complete
=== PAUSE TestAccSpatialAnchorsAccount_complete
=== CONT  TestAccSpatialAnchorsAccountDataSource_basic
=== CONT  TestAccSpatialAnchorsAccount_complete
=== CONT  TestAccSpatialAnchorsAccount_basic
--- PASS: TestAccSpatialAnchorsAccountDataSource_basic (105.96s)
--- PASS: TestAccSpatialAnchorsAccount_complete (111.01s)
--- PASS: TestAccSpatialAnchorsAccount_basic (111.88s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mixedreality	113.243s
```

Fixes #11799

### CODE REVIEW

**Comment by favoretti** [comment]
image

File: website/docs/d/spatial_anchors_account.html.markdown [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+---
+
+# azurerm_spatial_anchors_account
+
+Get information about an Azure Spatial Anchors Account.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
```
Comment (magodo) [inline website/docs/d/spatial_anchors_account.html.markdown:16]: Can we modify the example a bit to only includes the `data` block and the `output` block ([example](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/virtual_network))?

File: website/docs/d/spatial_anchors_account.html.markdown [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The ID of the Spatial Anchors Account.
+
+* `account_domain` - The domain of the Spatial Anchors Account.
+
+* `account_id` - Account ID of the Spatial Anchors Account.
```
Comment (magodo) [inline website/docs/d/spatial_anchors_account.html.markdown:43]:
```suggestion
* `account_id` - The account ID of the Spatial Anchors Account.
```

File: website/docs/r/spatial_anchors_account.html.markdown [hunk 3]
```diff
@@ -43,6 +43,10 @@ The following attributes are exported:
 
 * `id` - The ID of the Spatial Anchors Account.
 
+* `account_domain` - The domain of the Spatial Anchors Account.
+
+* `account_id` - Account ID of the Spatial Anchors Account.
```
Comment (magodo) [inline website/docs/r/spatial_anchors_account.html.markdown:48]:
```suggestion
* `account_id` - The account ID of the Spatial Anchors Account.
```

File: azurerm/internal/services/mixedreality/spatial_anchors_account_data_source_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+func TestAccSpatialAnchorsAccountDataSource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_spatial_anchors_account", "test")
+	r := SpatialAnchorsAccountDataSource{}
+
+	data.DataSourceTest(t, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("name").Exists(),
```
Comment (magodo) [inline azurerm/internal/services/mixedreality/spatial_anchors_account_data_source_test.go:23]: Can we further check the existance of the `account_id` and the `account_domain`? Also applies to the test cases for the managed resource.

File: azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go [hunk 5]
```diff
@@ -126,6 +136,8 @@ func resourceSpatialAnchorsAccountRead(d *schema.ResourceData, meta interface{})
 	if location := resp.Location; location != nil {
 		d.Set("location", azure.NormalizeLocation(*location))
 	}
+	d.Set("account_domain", resp.AccountDomain)
```
Comment (magodo) [inline azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go:139]: Persumably we should nil check for `resp.Properties` here?

File: azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go [hunk 6]
```diff
@@ -126,6 +136,8 @@ func resourceSpatialAnchorsAccountRead(d *schema.ResourceData, meta interface{})
...
+	d.Set("account_id", resp.AccountID)
```
Comment (magodo) [inline azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go:140]: Same as above.

File: azurerm/internal/services/mixedreality/spatial_anchors_account_data_source.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+		return fmt.Errorf("retrieving Spatial Anchors Account %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	d.SetId(id.ID())
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	d.Set("account_domain", resp.AccountDomain)
```
Comment (magodo) [inline azurerm/internal/services/mixedreality/spatial_anchors_account_data_source.go:77]: Persumably need to nil check for the `resp.Properties` here?

File: azurerm/internal/services/mixedreality/spatial_anchors_account_data_source.go [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+	d.Set("account_id", resp.AccountID)
```
Comment (magodo) [inline azurerm/internal/services/mixedreality/spatial_anchors_account_data_source.go:78]: Same as above.

**Review by magodo — CHANGES_REQUESTED** [review]
@favoretti Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

**Comment by favoretti** [comment]
```
$ TF_ACC=1 go test -v ./azurerm/internal/services/mixedreality -timeout=1000m
2021/05/24 09:36:19 [DEBUG] not using binary driver name, it's no longer needed
2021/05/24 09:36:19 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccSpatialAnchorsAccountDataSource_basic
=== PAUSE TestAccSpatialAnchorsAccountDataSource_basic
=== RUN   TestAccSpatialAnchorsAccount_basic
=== PAUSE TestAccSpatialAnchorsAccount_basic
=== RUN   TestAccSpatialAnchorsAccount_complete
=== PAUSE TestAccSpatialAnchorsAccount_complete
=== CONT  TestAccSpatialAnchorsAccountDataSource_basic
=== CONT  TestAccSpatialAnchorsAccount_complete
=== CONT  TestAccSpatialAnchorsAccount_basic
--- PASS: TestAccSpatialAnchorsAccountDataSource_basic (104.19s)
--- PASS: TestAccSpatialAnchorsAccount_basic (111.64s)
--- PASS: TestAccSpatialAnchorsAccount_complete (113.30s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mixedreality	114.963s
```

File: azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go (hunk 5 above)
Comment (magodo) [reply azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go]: This wants to be:

```go
if props := resp.SpatialAnchorsAccountProperties; props != nil {
  d.Set("account_domain", props.AccountDomain)
  d.Set("account_id", props.AccountID)
}
```

The reason is that the API might return response with `SpatialAnchorsAccountProperties` being absent, which will cause the dereference shortcut panic.

Comment (favoretti) [reply azurerm/internal/services/mixedreality/spatial_anchors_account_resource.go]: aaah, makes sense now. Fixing, thanks!

**Comment by favoretti** [comment]
```
$ TF_ACC=1 go test -v ./azurerm/internal/services/mixedreality -timeout=1000m
2021/05/24 11:21:59 [DEBUG] not using binary driver name, it's no longer needed
2021/05/24 11:22:00 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccSpatialAnchorsAccountDataSource_basic
=== PAUSE TestAccSpatialAnchorsAccountDataSource_basic
=== RUN   TestAccSpatialAnchorsAccount_basic
=== PAUSE TestAccSpatialAnchorsAccount_basic
=== RUN   TestAccSpatialAnchorsAccount_complete
=== PAUSE TestAccSpatialAnchorsAccount_complete
=== CONT  TestAccSpatialAnchorsAccountDataSource_basic
=== CONT  TestAccSpatialAnchorsAccount_complete
=== CONT  TestAccSpatialAnchorsAccount_basic
--- PASS: TestAccSpatialAnchorsAccountDataSource_basic (104.94s)
--- PASS: TestAccSpatialAnchorsAccount_basic (111.45s)
--- PASS: TestAccSpatialAnchorsAccount_complete (111.70s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mixedreality	113.732s
```

**Comment by favoretti** [comment]
image

**Comment by magodo** [comment]
@favoretti  LGTM :+1:

**Pull request merged**